### DISCUSSION TITLE
These carving cuts are very rough, what can I do to make them smoother?

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#118] [Help - Using SofaUnity]
Hello,
These carving cuts are very rough, what can I do to make them smoother?
Thanks!
![Unity_tYenLG2Bww](https://github.com/user-attachments/assets/b49cd602-80da-4f92-96cd-c991c8d71efc)

**Comment by epernod** [comment]
Hi,

carving is different from cutting. Carving remove element. If you want it smoother you need to refine more your mesh or to use a cutting manager.

Best regards

ps: Could you validate and close your other discussions if it has been answered. Thanks

**Comment by MingSongJiang** [comment]
Hello,
How can I further improve the grid? If there are too many grids, it cannot be read out.
For cutting, it will cut immediately after touching, but for gas cutting, it will not cut immediately after touching.
Thanks!

**Reply by epernod** [reply]
What is the current grid resolution? Are you generating a .msh file as a grid?

What is gas cutting?

Best regards,

**Reply by MingSongJiang** [reply]
I made the tetrahedron mesh very small, so the cutouts don’t look that sharp overall, but if you look closely, there are still many sharp triangles.
And the object can't be too big, otherwise if the mesh is dense, it will take up a lot of space. If it's over 2M, it will be difficult to load it in Unity.
I use .msh file.This size is almost enough to load
![image](https://github.com/user-attachments/assets/cdfb1cdb-f753-4ce6-995b-bf2bdbdc4f22)

**Comment by CherryJGit** [comment]
Hi,
Have you managed to fix this issue yet? I’m facing the exact same problem and tried your solutions, but the results don’t hold up under close inspection (e.g., visible artifacts, instability at the cut surface when zoomed in). Would you be willing to share any improved solutions or troubleshooting?

**Reply by epernod** [reply]
Hi, 
as I explained this is a carving method, not cutting one. So it means it removes some element, there will always be sharp edges. On way is to create a well refined mesh with regular meshing.
How is your mesh? and what type of simulation are you targettng?

**Reply by CherryJGit** [reply]
hi,
Thank you for your reply, I want to simulate the ablation of part of the human tissue, so I exported the tetrahedral mesh of MSH in GMSH, if the filling is too large, it will not load, but there will also be sharp edges, and it will not stand up to close inspection, and it will also cause performance problems. I'd like to implement it in Unity, is there any other way to do it than fine meshes, such as visual deception, etc

**Reply by epernod** [reply]
could you show the mesh at start and during process.
thank you

**Reply by CherryJGit** [reply]
Everything works fine, I want to make the part of it smoother and not sharper than it is now, fine mesh is one of the options I have come up with, but I wonder if there is any other way to solve it better.
thank you

**Reply by epernod** [reply]
ok good. I guess you are looking for cutting and not carving. Check this discussion: https://github.com/InfinyTech3D/SofaUnity/discussions/133

**Reply by CherryJGit** [reply]
I want to simulate a breast fat ablation procedure – outer skin layer for incision, inner adipose tissue for ablation. The current ablation boundaries lack smoothness; I need them to feel more like natural flesh ablation, not just geometric cuts. Looking for methods besides mesh refinement.
Best regards

**Reply by epernod** [reply]
Hi,
if you need incision in thin layer I would recommand to use tearing plugin: https://infinytech3d.com/examples-tearing/
If you need incision in volume you will need the MeshRefinement plugin: https://infinytech3d.com/examples-interactions/#5

The carving feature is removing elements without any subdivision. You can also combine it with remsehing: https://infinytech3d.com/examples-interactions/#4

we are working hard to make all those feature easy to use but as explained in the website. If you need to create complex simulation scenario, I would recommand to contact us for a quote. 

Best regards,
